spec/lib/promises.helper.spec.js

```diff
@@ -0,0 +1,136 @@
+         */
+        function cleanup() {
+            global.jasmine = oldJasmine;
+        }
+
+        oldJasmine = global.jasmine;
+
+        if (optionalMock) {
+            mock = optionalMock;
+        } else {
+            mock = mockRequire.reRequire(`../mock/jasmine-${version}-mock.js`)();
+        }
+
+        try {
+            global.jasmine = mock;
+            mockRequire.reRequire(filePath);
+        } catch (err) {
+            cleanup();
+            throw err;
+        }
+
+        cleanup();
+
+        return mock;
+    }
+    describe("check for promises working", () => {
+        beforeEach(() => {
+            check = 0;
+
+            return new Promise((resolve) => {
+                setTimeout(() => {
+                    check = 1;
+                    resolve();
+                }, 100);
+            });
+        });
+        it("waited for the beforeEach to finish", () => {
+            expect(check).toBe(1);
+
+            return new Promise((resolve) => {
+                setTimeout(() => {
+                    check = 2;
+                    resolve();
+                }, 100);
+            });
+        });
+        afterEach(() => {
+            expect(check).toBe(2);
+        });
+    });
+    describe("Version compatability:", () => {
+        describe("A Jasmine version that utitlize prototypal inheritance", () => {
+            var jasmineMock;
+
+            beforeEach(() => {
+                jasmineMock = versionTestHelper("../../lib/promises.helper.js", "prototypal").Env.prototype;
+            });
+            it("has patched methods", () => {
+                var methods;
+
+                methods = [
+                    "afterEach",
+                    "beforeEach",
+                    "iit",
+                    "it",
+                    "xit"
+                ];
+                testForPatching(jasmineMock, methods);
+            });
+        });
+        describe("A Jasmine version that doesn't utilize prototypal inheritance", () => {
+            var jasmineMock;
+
+            beforeEach(() => {
+                jasmineMock = versionTestHelper("../../lib/promises.helper.js", "nonprototypal").Env;
+            });
+            it("has patched methods", () => {
+                var methods;
+
+                methods = [
+                    "afterEach",
+                    "beforeEach",
+                    "fit",
+                    "it",
+                    "xit"
+                ];
+                testForPatching(jasmineMock, methods);
+            });
+        });
+    });
+    it("only patches for promises once", () => {
+        var jasmineCopy;
+
+        jasmineCopy = jasmine.then;
+        mockRequire.reRequire("../../lib/promises.helper.js");
+        expect(jasmineCopy).toBe(jasmine.then);
+    });
+    it("does not patch methods that already handle promises", (done) => {
+        mockRequire.reRequire("../../lib/promises.helper.js");
```

Comment:
This line is not necessary.  The `it` needs to have been patched before the body of this function is called, which it is.